File: Java/BinaryTrees/Medium/104-maximum-depth-of-binary-tree/maximum-depth-of-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int height;
    public void max(TreeNode root, int curr){
        if(root==null)return;
        height = Math.max(curr,height);
        max(root.left, curr+1);
        max(root.right,curr+1);
    }
    public int maxDepth(TreeNode root) {
        max(root,1);
        return height;
    }
}

File: Python/LinkedList/104-maximum-depth-of-binary-tree/maximum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def search(self, root,ans,cnt):
        if root is None:
            return     
            
        ans[0] = max(ans[0],cnt)
        
        
        if root.left:
            self.search(root.left, ans, cnt+1)
        
        if root.right:
            self.search(root.right, ans, cnt+1)

        
        
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        ans = [0]
        self.search(root, ans, 1)

        return ans[0]